rasa/shared/core/events.py

```diff
@@ -1211,15 +1223,22 @@ def __init__(
 
         super().__init__(timestamp, metadata)
 
-    def __str__(self) -> Text:
+    def __repr__(self) -> Text:
+        """Returns event as string for debugging."""
         return "ActionExecuted(action: {}, policy: {}, confidence: {})".format(
             self.action_name, self.policy, self.confidence
         )
 
+    def __str__(self) -> Text:
+        """Returns event as human readable string."""
+        return self.action_name or self.action_text
+
     def __hash__(self) -> int:
-        return hash(self.action_name)
+        """Returns unique hash for action event."""
+        return hash(self.action_name or self.action_text)
 
-    def __eq__(self, other) -> bool:
+    def __eq__(self, other: Any) -> bool:
+        """Checks if object is equal to another."""
         if not isinstance(other, ActionExecuted):
             return False
```

Comment:
I know not changed in this PR, but as we were discussing dunder methods above...
When the other is a different class we should actually _return_ `NotImplemented` (not raise it, or return `False`).
This allows the interpreter to do clever things like try the reflected \_\_eq\_\_ operation.

Reply:
thanks! TIL!